### PR TITLE
cleanup cancel

### PR DESCRIPTION
**Pull request by bekzod**

### CODE REVIEW

File: lib/index.ts [hunk 1]
```diff
@@ -611,10 +611,16 @@ export default class Backburner {
...
   public cancel(timer?) {
+    if (!timer) { return; }
     let timerType = typeof timer;
 
-    if (timer && timerType === 'object' && timer.queue && timer.method) { // we're cancelling a deferOnce
-      return timer.queue.cancel(timer);
+    if (timerType === 'object') {
+      if (timer.queue && timer.method) { // we're cancelling a deferOnce
+        return timer.queue.cancel(timer);
+      } else if (Object.prototype.toString.call(timer) === '[object Array]') { // we're cancelling a throttle or debounce
```
Comment (stefanpenner) [inline lib/index.ts]: IsArray (or is the form the new PR?)

Comment (bekzod) [inline lib/index.ts]: will rebase once isArray PR merged

**Comment by bekzod** [comment]
this doubled op/s for `cancel` in benchmark

File: lib/index.ts (hunk 1 above)
Comment (stefanpenner) [inline lib/index.ts]: https://github.com/BackburnerJS/backburner.js/pull/214

**Comment by bekzod** [comment]
this can be merged I think :P

**Pull request merged**